src/gt4py/next/program_processors/runners/dace_iterator/itir_to_sdfg.py

```diff
@@ -281,6 +330,212 @@ def visit_StencilClosure(
 
         return closure_sdfg
 
+    def visit_ScanStencilClosure(
+        self,
+        node: itir.StencilClosure,
+        array_table: dict[str, dace.data.Array],
+        closure_domain: tuple[tuple[str, tuple[ValueExpr, ValueExpr]], ...],
+    ) -> tuple[dace.SDFG, dict[str, str | dace.subsets.Subset], list[str], int]:
+        # extract scan arguments
+        def get_scan_args() -> tuple[Literal, Literal]:
+            stencil_fobj = cast(FunCall, node.stencil)
+            return cast(Literal, stencil_fobj.args[1]), cast(Literal, stencil_fobj.args[2])
+
+        # select the scan dimension based on program argument for column axis
+        def get_scan_dim() -> tuple[str, int, ScalarType]:
+            assert isinstance(node.output, SymRef)
+            field_type = cast(FieldType, self.storages[node.output.id])
+            assert isinstance(self.column_axis, Dimension)
+            return (
+                self.column_axis.value,
+                field_type.dims.index(self.column_axis),
+                field_type.dtype,
+            )
```

Comment:
I'd extract these to the enclosing module as private functions, or if not possible, find other ways to break up the enclosing function. These utility functions are often good candidates for pure functions that have no side effects, thus always return the same results for the same inputs. Pure function are the best for testing, but by making them nested function and relying on capturing as opposed to function parameters for inputs you eliminate this ease of testing.

Reply:
Good point about the testability aspect, I will try to make pure functions in a nice way.